Cython/Compiler/Pipeline.py

```diff
@@ -45,7 +45,7 @@ def parse(source_desc):
         return tree
     return parse
 
-def generate_pyx_code_stage_factory(options, result, cimport_from_pyx=False):
+def generate_pyx_code_stage_factory(options, result, cimport_from_pyx=Options.cimport_from_pyx):
```

Comment:
This nails the option at import time, regardless of the value at runtime. If users change it after importing Cython, this will fail to produce the intended result.

It's apparently needed across the whole compiler run, from parse to code generation. Thus, it shouldn't be set just here and during parsing. We should change the setting for the whole run of the `SharedModule` pipeline. You can add a pipeline function as reset step after the code generation.